src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -42,13 +45,26 @@
   private boolean isLoggedIn;
   private String url;
   private String email;
+  private ArrayList<Entity> moderatingOrgs = new ArrayList<Entity>();
 
-  public GivrUser(String id, boolean isMaintainer, boolean isLoggedIn, String url, String email) {
+  public GivrUser(String id, boolean isMaintainer, boolean isLoggedIn, String url, String email, ArrayList<Entity> moderatingOrgs) {
     this.id = id;
     this.isMaintainer = isMaintainer;
     this.isLoggedIn = isLoggedIn;
     this.url = url;
     this.email = email;
+    this.moderatingOrgs = moderatingOrgs;
+
+    if (!this.isMaintainer) {
+        setModeratingOrgs();
+        ArrayList<Entity> orgList = getModeratingOrgs();
+        if (orgList.size() > 0) {
+          for (int i = 0; i < orgList.size(); i++) {
+            OrganizationUpdater organizationUpdater = new OrganizationUpdater(orgList.get(i));
+            organizationUpdater.updateInvitedModerator(this);
```

Comment:
Noted! I'm now wondering whether it would be okay to just call the setModeratingOrgs() within GivrUser's constructor, and within setModeratingOrgs(), update each Organization (that the GivrUser is a moderator of) using OrganizationUpdater. Somehow, my code needs to accomplish updating Organization entities in the Datastore.

Reply:
Why does it need to do that in the constructor though? In particular, the new function: "setModeratingOrgsAndUpdateOrgs" has a great name, as it's obvious what it does. But: why is that the function that exists instead of "setModeratingOrgs" + "UpdateOrgs"

(And why is "updateOrgs" a function on giveruser and not on orgupdater?

Reply:
Yes, we do have an updateInvitedModerator method in OrgUpdater which is called in the setModeratingOrgsAndUpdateOrgs.

The reason why we need update to be invoked within GivrUser is because each Organization that the user is a moderator of needs: 1) remove the user's email from the invitedModerators list of that specific Organization, and 2) add the user's ID to the moderatorList of that Organization.

I will refactor the setModeratingOrgs and UpdateOrgs to be separate methods, but I'm not sure if I should be calling updateOrgs within the GivrUser's constructor. Somehow, this needs to be called whenever a user logs in, and this could be in the getCurrentLoggedInUser().

Reply:
Would everything work for now if we skipped 1,2?

Reply:
Just responding to this comment to document that we have discussed and came to a solution for this issue: Call setting moderating orgs and update organizations (invited moderator list -> moderator list) within Authenticate Servlet, as every user (both logged in & not logged in) calls it.